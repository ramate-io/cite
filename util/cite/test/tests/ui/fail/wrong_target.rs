// Test that citations on truly invalid targets fail to compile

use cite_util::{cite, mock::MockSource};

// This should fail - can't cite a const item
#[cite(MockSource::same("content"))]
const INVALID_TARGET: i32 = 42;

fn main() {}
