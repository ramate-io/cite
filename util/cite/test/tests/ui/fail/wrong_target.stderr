error: cite attribute can only be applied to functions, structs, traits, impl blocks, or modules
 --> tests/ui/fail/wrong_target.rs:7:1
  |
7 | const INVALID_TARGET: i32 = 42;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `mock::MockSource`
 --> tests/ui/fail/wrong_target.rs:3:23
  |
3 | use cite_util::{cite, mock::MockSource};
  |                       ^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
